packages/medusa/src/services/totals-new.ts

```diff
@@ -0,0 +1,692 @@
+import {
+  ITaxCalculationStrategy,
+  TaxCalculationContext,
+  TransactionBaseService,
+} from "../interfaces"
+import { EntityManager } from "typeorm"
+import {
+  Discount,
+  DiscountRuleType,
+  GiftCard,
+  LineItem,
+  LineItemTaxLine,
+  Region,
+  ShippingMethod,
+  ShippingMethodTaxLine,
+} from "../models"
+import { TaxProviderService, TotalsService } from "./index"
+import { LineAllocationsMap } from "../types/totals"
+import TaxInclusivePricingFeatureFlag from "../loaders/feature-flags/tax-inclusive-pricing"
+import { FlagRouter } from "../utils/flag-router"
+import { calculatePriceTaxAmount, isDefined } from "../utils"
+import { MedusaError } from "medusa-core-utils"
+
+type LineItemTotals = {
+  unit_price: number
+  quantity: number
+  subtotal: number
+  tax_total: number
+  total: number
+  original_total: number
+  original_tax_total: number
+  tax_lines: LineItemTaxLine[]
+  discount_total: number
+}
+
+type ShippingMethodTotals = {
+  price: number
+  tax_total: number
+  total: number
+  subtotal: number
+  original_total: number
+  original_tax_total: number
+  tax_lines: ShippingMethodTaxLine[]
+}
+
+type InjectedDependencies = {
+  totalsService: TotalsService
+  taxProviderService: TaxProviderService
+  taxCalculationStrategy: ITaxCalculationStrategy
+  featureFlagRouter: FlagRouter
+}
+
+export default class TotalsNewService extends TransactionBaseService {
+  protected readonly manager_: EntityManager
+  protected readonly transactionManager_: EntityManager | undefined
+
+  protected readonly totalsService_: TotalsService
+  protected readonly taxProviderService_: TaxProviderService
+  protected readonly featureFlagRouter_: FlagRouter
+  protected readonly taxCalculationStrategy_: ITaxCalculationStrategy
+
+  constructor({
+    totalsService,
+    taxProviderService,
+    featureFlagRouter,
+    taxCalculationStrategy,
+  }: InjectedDependencies) {
+    super(arguments[0])
+
+    this.totalsService_ = totalsService
+    this.taxProviderService_ = taxProviderService
+    this.featureFlagRouter_ = featureFlagRouter
+    this.taxCalculationStrategy_ = taxCalculationStrategy
+  }
+
```

Comment:
suggestion: could we overload `getLineItemTotals` so that we have:
```ts
async getLineItemTotals(items: LineItem[], config: LineItemTotalsConfig): Promise<{ [lineItemId: string]: LineItemTotals }>

async getLineItemTotals(item: LineItem, config: LineItemTotalsConfig): Promise<LineItemTotals>
```

It is confusing that two methods do a similar thing one just takes an array? 

Also wouldn't it make sense to align the two implementations so getLineItemTotals basically is the same as:
```
getLineItemTotals([item], ...)
```

Reply:
I can just make the `getLineItemTotals` protected and only export the `getLineItemsTotals` as part of the public API. The idea was to separate the calculation from the rest in order to make it  easier to read and have smaller function. wdyt? the `getLineItemTotals` only take care of the calculation

Reply:
I did make the other method protected and only the main one it public, which accept a collection of items and that can be only one item.